src/app/services/session-storage/session-storage.service.ts

```diff
@@ -7,9 +7,14 @@ export class SessionStorageService {
 
   /**
    * Get an item from the session storage.
+   * If remove is true, the item will be removed once read
    */
-  public getItem(key: string): string {
-    return sessionStorage.getItem(key);
+  public getItem(key: string, remove: boolean = false): string {
```

Comment:
I know it might be questioned why I have added the option.

I want the value to removed after it has been read. I could do this by calling getItem() and then removeItem() in my calling code. But that makes me nervous. I want the read and delete to be done in one, atomic request. I could have added a new method getAndDeleteItem() but that seemed an odd thing to do.

Reply:
Logically, no problem with this. My only slight concern is with the name of the property as I don't think it's _entirely_ clear what the behaviour will be. Renaming the `remove` parameter to `removeAfterRead` would make it much clearer.